Chainguard Repository for Java (#3316)
chainguard-dev/edu: content/chainguard/libraries/java/build-configuration.md

#### Configure direct access

If you are not using a repository manager at your organization, you can
configure access to the Chainguard Libraries for Java repository directly. If [upstream fallback](/chainguard/libraries/overview/#upstream-fallback-and-controls) is enabled for your organization, the `https://libraries.cgr.dev/java/` repository can serve both Chainguard-built artifacts and eligible upstream Maven Central artifacts through the same endpoint. If upstream fallback is not enabled, continue to configure Maven Central or your Maven Central proxy after the Chainguard repository, as shown in the following example.

If you are participating in the beta for CVE remediation, include the `https://libraries.cgr.dev/java-remediated/` repository first.

> If you are using direct access with the Chainguard Repository and you want Chainguard policy controls to apply consistently, configure Chainguard as a global Maven mirror. Without a global mirror, Maven can fall back to its built-in Maven Central definition when Chainguard reports a dependency as unavailable, bypassing the expected cooldown behavior for artifacts.

Reviewer: ```suggestion
If you are using direct access with the Chainguard Repository and you want Chainguard policy controls to apply consistently, configure Chainguard as a global Maven mirror. Without a global mirror, Maven can fall back to its built-in Maven Central definition when Chainguard reports a dependency as unavailable, bypassing the policy and malware scanning controls provided by Chainguard.
```

> The following `~/.m2/settings.xml` configures direct access with Chainguard's remediated Java repository as

Reviewer: Flagging a comment to update this once we verify with vanguard testing what the right settings.xml config should look like.

#### Configure direct access

If you are not using a repository manager at your organization, you can

> configure access to the Chainguard Libraries for Java repository directly. If [upstream fallback](/chainguard/libraries/overview/#upstream-fallback-and-controls) is enabled for your organization, the `https://libraries.cgr.dev/java/` repository can serve both Chainguard-built artifacts and eligible upstream Maven Central artifacts through the same endpoint. If upstream fallback is not enabled, continue to configure Maven Central or your Maven Central proxy after the Chainguard repository, as shown below.

Reviewer 2: ```suggestion
configure access to the Chainguard Libraries for Java repository directly. If [upstream fallback](/chainguard/libraries/overview/#upstream-fallback-and-controls) is enabled for your organization, the `https://libraries.cgr.dev/java/` repository can serve both Chainguard-built artifacts and eligible upstream Maven Central artifacts through the same endpoint. If upstream fallback is not enabled, continue to configure Maven Central or your Maven Central proxy after the Chainguard repository, as shown in the following example.
```

'Below' can be a little ambiguous

> If you are participating in the beta for CVE remediation, include the `https://libraries.cgr.dev/java-remediated/` repository first.

Reviewer: will need to confirm if this will be ready before launch - but we will have an option to prioritize /java-remediated/ first within the repo

Reviewer 3: Just fyi that URL is part of the video that will be released during innovation week

Reviewer: we will most likely add this as a follow-up post innovation week

            <enabled>false</enabled>
          </snapshots>
        </repository>

>         <repository>

Reviewer 3: This must be kept in place and set to an invalid URL with fallback .. or to a valid URL if we want it live and acting last.

`pluginRepositories` to individual project `pom.xml` files. Authentication
details must remain within the settings file.

If your `settings.xml` is using credentials set as environment variables, ensure the variables are exported.

> To explicitly remove access to Maven Central, include the following in your `settings.xml`:

Reviewer 3: This makes it sound like its optional. That is NOT the case. The definition is built into the Maven binary itself and is always there. If you want to guarantee that there is no fallback to Maven Central you MUST include this
